xarray/core/variable.py

```diff
@@ -376,29 +387,127 @@ def _item_key_to_tuple(self, key):
         else:
             return key
 
+    def _broadcast_indexes(self, key):
+        """
+        Parameters
+        -----------
+        key: One of
+            array
+            a mapping of dimension names to index.
+
+        Returns
+        -------
+        dims: Tuple of strings.
+            Dimension of the resultant variable.
+        indexers: list of integer, array-like, or slice. This is aligned
+            along self.dims.
+        """
+        key = self._item_key_to_tuple(key)  # key is a tuple
+        # key is a tuple of full size
+        key = indexing.expanded_indexer(key, self.ndim)
+        if all(isinstance(k, BASIC_INDEXING_TYPES) for k in key):
+            return self._broadcast_indexes_basic(key)
+
+        # Detect it can be mapped as an outer indexer
+        # If all key is unlabelled, or
```

Comment:
It looks like both "labelled" and "labeled" are correct spellings, but elsewhere in xarray we use the American spelling "labeled" so please stick with that (here and below).